allennlp/training/trainer.py

```diff
@@ -451,6 +455,18 @@ def _get_metrics(self, total_loss: float, num_batches: int, reset: bool = False)
         metrics["loss"] = float(total_loss / num_batches) if num_batches > 0 else 0.0
         return metrics
 
+    def _get_batch_size(self, batch: Union[Dict, torch.Tensor]) -> int:
+        """
+        Returns the size of the batch dimension. Assumes a well-formed batch,
+        returns 0 otherwise.
+        """
+        if isinstance(batch, torch.Tensor):
+            return batch.size()[0] # type: ignore
```

Comment:
`return batch.size(0)` might alleviate that type warning

Reply:
Changed. But I'm still getting:
`allennlp/training/trainer.py:464: error: Item Dict[Any, Any] of "Union[Dict[Any, Any], Any]" has no attribute "size"`
Unclear why it's not using the `torch.Tensor` type correctly.